src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -524,5 +529,32 @@ private async Task RemoveContainerNetworkAsync(IExecutionContext executionContex
             // Remove docker network from env
             executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, null);
         }
+
+        private async Task ContainerHealthcheck(IExecutionContext executionContext, ContainerInfo container)
+        {
+            string healthCheck = "--format=\"{{if .Config.Healthcheck}}{{print .State.Health.Status}}{{end}}\"";
+            string serviceHealth = await _dockerManger.DockerInspect(context: executionContext, dockerObject: container.ContainerId, options: healthCheck);
+            if (string.IsNullOrEmpty(serviceHealth))
+            {
+                // Container has no HEALTHCHECK
+                return;
+            }
+            TimeSpan backoff = TimeSpan.FromSeconds(2);
+            while (string.Equals(serviceHealth, "starting", StringComparison.OrdinalIgnoreCase))
+            {
+                executionContext.Output($"{container.ContainerNetworkAlias} service is starting, waiting {backoff.Seconds} seconds before checking again.");
+                Thread.Sleep(backoff);
```

Comment:
never use Thread.Sleep() in async/await Task. :D
try use `BackoffTimerHelper` to get timespan.

Reply:
you can find `BackoffTimerHelper ` in this file. :)